app/Http/Controllers/ProfileController.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\PublisherProfileRequest;
+use App\Http\Requests\UserProfileRequest;
+use App\Models\Publisher;
+use App\Models\User;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Config;
+
+class ProfileController extends Controller
+{
+	public function index()
+	{
+		if(Auth::check()) {
+			if(Auth::user()->role == Config::get('role.user')) {
+				$user = Auth::user()->user;
+				if($user == null)
+					$user = new User();
+
+				return view('profile')->with('user', $user);
+			} else {
+				$publisher = Auth::user()->publisher;
+				if($publisher == null)
+					$publisher = new Publisher();
+
+				return view('profile')->with('publisher', $publisher);
+			}
+		} else {
+			return redirect()->route('login');
+		}
+	}
+
+	public function updateUser(UserProfileRequest $request)
+	{
+		$user = User::updateOrCreate(
+			['account_id' => Auth::user()->id],
+			[ 
+				'name' => $request->name,
+				'birthday' => $request->birthday,
+				'address' => $request->address,
+				'phone' => $request->phone,
+			]
+		);
+
+		return redirect()->back()->with('message', trans('text.profile.updated'));
+	}
+
+	public function updatePublisher(PublisherProfileRequest $request)
+	{
+		$publisher = Publisher::updateOrCreate(
+			['account_id' => Auth::user()->id],
+			[
+				'name' => $request->name,
+				'address' => $request->address,
+				'phone' => $request->phone,
```

Comment:
đọan này nên thay băng:
```suggestion
$data = $request->all();
$data['account_id'] = Auth::id();
Publisher::updateOrCreate($data);
```

Reply:
```suggestion
		$publisher = Publisher::updateOrCreate(
			['account_id' => Auth::user()->id],
			$request->all()
```